### PR TITLE
feat(metrics): emit metric extraction rules

### LINKED ISSUE #72851: Project config extraction rules generation

### PR DESCRIPTION
**Pull request by obostjancic**
- closes: #72851

### CODE REVIEW

File: src/sentry/snuba/metrics/extraction.py [hunk 1]
```diff
@@ -1690,9 +1691,11 @@ class SearchQueryConverter:
     The converter can be used exactly once.
     """
 
-    def __init__(self, tokens: Sequence[QueryToken]):
+    def __init__(self, tokens: Sequence[QueryToken], field_mapper: Any = _map_field_name):
```
Comment (vgrozdanic) [inline src/sentry/snuba/metrics/extraction.py:1694]: Should the type of `field_mapper` be `Callable[[str], str]`?

Comment (obostjancic) [reply src/sentry/snuba/metrics/extraction.py:1694]: Good catch, thanks!

**Review by shellmayr — APPROVED** [review]
LGTM 🚢

File: src/sentry/snuba/metrics/extraction.py (hunk 1 above)
Comment (obostjancic) [reply src/sentry/snuba/metrics/extraction.py]: [727670f](https://github.com/getsentry/sentry/pull/72997/commits/727670f258271bfbc52df750d508c32e3da2edb4)

**Review by vgrozdanic — APPROVED** [review]

File: tests/sentry/relay/config/test_metric_extraction.py [hunk 2]
```diff
@@ -2133,3 +2134,45 @@ def test_get_current_widget_specs(
...
         specs = get_current_widget_specs(default_project.organization)
     assert specs == expected
+
+
+@django_db_all
+def test_get_span_attribute_metrics(default_project: Project) -> None:
+    rules = [
+        {
+            "spanAttribute": "span.duration",
+            "mri": "d:custom/span.duration",
```
Comment (shellmayr) [inline tests/sentry/relay/config/test_metric_extraction.py:2144]: nit: the MRI is missing a unit here

**Pull request merged**